### ISSUE TITLE
Move the service worker (or its scope), so that it catches all the request from the iframe, and not the ones from the UI

### ISSUE THREAD

**Issue by mossroy** [kiwix/kiwix-js#138]
It's a blocker for some ZIM files (for example wikipedia_fr_all_09_2012.zim), which use absolute URLs like /A/Article.html.
And the service-worker.js file should be placed with the other javascript files (if possible)

**Comment by mossroy** [comment]
@kelson42 tells me that the norm is now to have relative URLs everywhere (since the end of 2014).
So this issue is not crucial, but it would be much cleaner.

**Comment by kelson42** [comment]
I would not recommend to handle cases which are not respecting the ZIM specification. That only makes the code more complicated.... for no added value. I strongly recommend to close this ticket as wont-fix.

**Comment by mossroy** [comment]
No.
The goal here is not to handle faulty ZIM files, but to make the ServiceWorker only handle what it should handle.
Currently, it technically catches ALL the HTTP requests made by the browser, including the ones to load the resource of kiwix-js itself (like fetching www/js/app.js).

There is a regular expression at the beginning of the ServiceWorker that checks if the URL matches a ZIM URL. It ignores the rest, so that everything works.

But it would be much better if we managed to make the ServiceWorker catch only the requests that come from the iframe.
Now that we implemented #534, we might catch only the URLs with our prefix.
The problem is that this prefix is the ZIM file name, which we do not know when the ServiceWorker is initialized.

A solution might be to change our prefix : instead of having only the ZIM file name, we might use something like `"articleContent/" + zimFileName`, and make the ServiceWorker catch only the URLs starting with "articleContent/"
This should not be very difficult.
